src/v/cluster/health_monitor_backend.cc

```diff
@@ -827,4 +827,28 @@ bool health_monitor_backend::does_raft0_have_leader() {
     return _raft0->get_leader_id().has_value();
 }
 
+void health_monitor_backend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Backend");
+    // Exlicitly stop/wipe state here instead of calling stop() because stop()
+    // will also close the gate and break the mutex
+    _raft_manager.local().unregister_leadership_notification(
+      _leadership_notification_handle);
+
+    abort_current_refresh();
```

Comment:
This calls `_refresh_request->abort();` , then eventually will set a promise to a value of:
```
done.set_value(errc::leadership_changed);
```
`errc::leadership_changed` would be an incorrect error code to fulfill the promise with because invocation of `reset()` is the reason why the promise was fulfilled. Maybe we can modify the class so that another error code may be called on abort due to this type of cancellation.

Reply:
I see your point. `_refresh_request` is auto-reset  when the health cache is refreshed anyways. From `health_monitor_backend::refresh_cluster_health_cache`:
```cpp
    _refresh_request = ss::make_lw_shared<abortable_refresh_request>(
      *leader_id, std::move(holder), std ::move(units));
```

I'm thinking that a better way to "reset" this thing is to set `finished = true` and then release the shared pointer. Then let `refresh_cluster_health_cache` recreate it/